src/handler/authentication.rs

```diff
@@ -140,3 +141,26 @@ pub async fn login(
 
     Ok((StatusCode::OK, headers))
 }
+
+pub async fn logout(
+    State(state): State<Repository>,
+    TypedHeader(cookie): TypedHeader<Cookie>,
+) -> Result<impl IntoResponse, StatusCode> {
+    let session_id = cookie.get("session_id").ok_or(StatusCode::UNAUTHORIZED)?;
+
+    state
+        .delete_session(session_id)
+        .await
+        .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?
+        .ok_or(StatusCode::UNAUTHORIZED)?;
+
+    let mut headers = HeaderMap::new();
+    headers.insert(
+        SET_COOKIE,
+        "session_id=; HttpOnly; SameSite=Lax; Max-Age=-1"
+            .parse()
+            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?,
+    );
+
+    Ok((StatusCode::OK, headers))
```

Comment:
openapiだとNO_CONTENTです。
```suggestion
    Ok((StatusCode::NO_CONTENT, headers))
```